go/vt/vtctl/workflow/traffic_switcher.go

```diff
@@ -1203,6 +1220,278 @@ func (ts *trafficSwitcher) isSequenceParticipating(ctx context.Context) (bool, e
 	return sequenceFound, nil
 }
 
+// getTargetSequenceMetadata returns a map of sequence metadata keyed by the
+// backing sequence table name. If the target keyspace has no tables
+// defined that use sequences for auto_increment generation then a nil
+// map will be returned.
+func (ts *trafficSwitcher) getTargetSequenceMetadata(ctx context.Context) (map[string]*sequenceMetadata, error) {
+	vschema, err := ts.TopoServer().GetVSchema(ctx, ts.targetKeyspace)
+	if err != nil {
+		return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "failed to get vschema for target keyspace %s: %v",
+			ts.targetKeyspace, err)
+	}
+	if vschema == nil || vschema.Tables == nil || len(vschema.Tables) == 0 { // Nothing to do
```

Comment:
Assuming `vschema.Tables` is a slice,

```go
if vschema == nil || len(vschema.Tables) == 0
```

Is sufficient because a nil check is not needed to guard `len` on a slice.

Reply:
Yeah, this got left in I think at some point. Will remove the slice nil check. Thanks!

Reply:
vschema.Tables is not a slice, but rather `map[string]*vschemapb.Table`.

Reply:
Also fine! https://go.dev/play/p/Nm8CZMdab1C